erdjs-snippets/lending-pool/lendingPoolInteractor.ts

```diff
@@ -0,0 +1,348 @@
+            .withChainID(this.networkConfig.ChainID);
+
+        // Let's check the interaction, then build the transaction object.
+        let transaction = interaction.check().buildTransaction();
+
+        // Let's sign the transaction. For dApps, use a wallet provider instead.
+        await user.signer.sign(transaction);
+
+        // The contract address is deterministically computable:
+        let address = SmartContract.computeAddress(transaction.getSender(), transaction.getNonce());
+        this.contract.setAddress(address);
+
+        // Let's broadcast the transaction and await its completion:
+        await this.networkProvider.sendTransaction(transaction);
+        let transactionOnNetwork = await this.transactionWatcher.awaitCompleted(transaction);
+
+        // In the end, parse the results:
+        let { returnCode } = this.resultsParser.parseOutcome(transactionOnNetwork, interaction.getEndpoint());
+        console.log(`LendingPoolInteractor.addLiquidityPool(): contract = ${address}`);
+
+        return returnCode;
+    }
+
+
+    async issueLend(user: ITestUser, tokenIdentifier: string): Promise<ReturnCode> {
+        console.log(`LendingPoolInteractor.issueLend(): address = ${user.address}`);
+
+        // Prepare the interaction
+        let interaction = <Interaction>this.contract.methods
+            .issueLendToken([tokenIdentifier, tokenIdentifier.split("-")[0]])
+            .withGasLimit(120000000)
+            .withValue(50000000000000000)
+            .withNonce(user.account.getNonceThenIncrement())
+            .withChainID(this.networkConfig.ChainID);
+
+        // Let's check the interaction, then build the transaction object.
+        let transaction = interaction.check().buildTransaction();
+
+        // Let's sign the transaction. For dApps, use a wallet provider instead.
+        await user.signer.sign(transaction);
+
+        // The contract address is deterministically computable:
+        let address = SmartContract.computeAddress(transaction.getSender(), transaction.getNonce());
+        this.contract.setAddress(address);
+
+        // Let's broadcast the transaction and await its completion:
+        await this.networkProvider.sendTransaction(transaction);
+        let transactionOnNetwork = await this.transactionWatcher.awaitCompleted(transaction);
+
+        // In the end, parse the results:
+        let { returnCode } = this.resultsParser.parseOutcome(transactionOnNetwork, interaction.getEndpoint());
+        return returnCode;
+    }
+
+    async issueBorrow(user: ITestUser, tokenIdentifier: string): Promise<ReturnCode> {
+        console.log(`LendingPoolInteractor.issueBorrow(): address = ${user.address}`);
+
+        // Prepare the interaction
+        let interaction = <Interaction>this.contract.methods
+            .issueBorrowToken([tokenIdentifier, tokenIdentifier.split("-")[0]])
+            .withValue(50000000000000000)
+            .withGasLimit(120000000)
```

Comment:
Less gas limit, perhaps? (here, and above / below etc.) Isn't it too much, 120 mil?

Reply:
It's a bit of guessing, I tried with lower gas limit [1], but it wasn't enough. It seems it needs around 90 mil [2]

[1] https://devnet-explorer.elrond.com/transactions/3a5fd1eab0392e3bff6aa2aa90110a9d8fdbbe9756ca4d0d279c2f65aef7269d
[2] https://devnet-explorer.elrond.com/transactions/bcf61a13d55d9c4bde7275eec07df7c20719eaba0c6aa8135a2692779047a35f